latex/slides/07_arrays.tex

```diff
@@ -350,40 +348,58 @@ \subsection{}
 		\item \textit{stdin} is the standard input
 	\end{itemize}
 \end{frame}
-\begin{frame}[fragile]{Array of arrays}
+\begin{frame}[fragile]{Passing arrays}
+	An array is passed by its identifier, no big deal. The real question is, how a function is defined, that takes an array as argument.
+	\begin{lstlisting}[numbers=none]
+int foo(char arr[]) {
+	printf("%d\n", arr[0]);
+}
+\end{lstlisting}
+	\begin{itemize}
+		\item \textbf{But:} If you want to get the size of the array in this function, you'll get a false value. The reason for this is discussed later.
+		\item So you may want to pass the size as a second argument
+		\item This prevents out-of-bounds problems
+	\end{itemize}
+\end{frame}
+%
+\begin{frame}[fragile]{Arrays of arrays}
 	\begin{itemize}
-		\item Every sub-array is of same type
+		\item Every sub-array is of the same type
+		\item All dimensions apart from the first \underline{must} be specified
 		\item Declaration:
 		\begin{lstlisting}[numbers=none]
 char arr[3][4];
-  /* an array containing 3 arrays containing 4 chars */
+/* an array containing 3 arrays containing 4 chars */
 \end{lstlisting}
 		\item Assignment:
 		\begin{lstlisting}[numbers=none]
-arr[0][0] = 'f'; /* first char of first array is 'f' */
-arr[1] = "bar";
+arr[0][0] = 'f'; /* first element of arr[0] is 'f' */
+arr[1] = "bar";  /* first element of arr is "bar" */
 arr[2] = {'b', 'o', 'z', '\0'};
 \end{lstlisting}
 		\item Definition:
 		\begin{lstlisting}[numbers=none]
-char arr[][] = {{'f', 'o', 'o', '\0'},
-				{'b', 'a', 'r', '\0'},
-				{'b', 'o', 'z', '\0'}};
+char arr[][4] = {{'f', 'o', 'o', '\0'},
+			     {'b', 'a', 'r', '\0'}};
 \end{lstlisting}
 	\end{itemize}
 \end{frame}
-\begin{frame}[fragile]{Passing arrays}
-	An array is passed by its name, no big deal. The real question is, how a function is defined, that takes an array as argument.
-	\begin{lstlisting}[numbers=none]
-int foo(char arr[]) {
-	printf("%d\n", arr[0]);
+%
+\begin{frame}[fragile]{Passing arrays of arrays}
+	\begin{itemize}
+		\item You can also pass arrays of arrays to functions
+		\item Once again, we need to specify the dimensions in the declaration
+		\item One way to solve this are dimension arguments in \underline{front} of the array
```

Comment:
I don't get what you want to say by this. Why must the dimensions be passed in front of the array?

The only new information this slides gives is that multidimensional arrays in function calls are handled like in array definitions: you have to specify all dimensions but the first.